lib/Dialect/FIRRTL/Import/FIRAnnotations.cpp

```diff
@@ -0,0 +1,181 @@
+//===- FIRAnnotations.cpp - .fir to FIRRTL dialect parser -----------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// Provide utilities related to dealing with Scala FIRRTL Compiler Annotations.
+//
+//===----------------------------------------------------------------------===//
+
+#include "circt/Dialect/FIRRTL/FIRAnnotations.h"
+
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/OperationSupport.h"
+
+using namespace circt;
+using namespace firrtl;
+
+namespace circt {
+namespace firrtl {
+
+/// Deserialize a JSON value into a Target-keyed array of Annotations
+/// (represented as DictionaryAttrs).
+bool fromJSON(llvm::json::Value &value,
```

Comment:
You can remove the namespace scoping on line 21/22 and declare this as:

bool circt::firrtl::fromJSON(llvm::json::Value &value, ...

Reply:
Flattened this out as suggested.